From a21030b7bbda26dcb337b599a5de0029834d5714 Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Sun, 7 May 2023 07:10:39 +0200
Subject: [PATCH] Add two spidevs max31865

---
 .../dts/sun8i-h2-plus-bananapi-m2-zero.dts    | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-h2-plus-bananapi-m2-zero.dts b/arch/arm/boot/dts/sun8i-h2-plus-bananapi-m2-zero.dts
index 938979e29da9..69a5d83f19f3 100644
--- a/arch/arm/boot/dts/sun8i-h2-plus-bananapi-m2-zero.dts
+++ b/arch/arm/boot/dts/sun8i-h2-plus-bananapi-m2-zero.dts
@@ -342,3 +342,28 @@ &usbphy {
 	 */
 	status = "okay";
 };
+
+&pio {
+	spi0_cs_pins: spi0-cs-pins {
+		pins = "PC3, PC4";
+		function = "gpio_out";
+	};
+};
+
+&spi0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi0_cs_pins>, <&spi0_pins>;
+	cs-gpios = <&pio 2 3 GPIO_ACTIVE_LOW>, <&pio 2 4 GPIO_ACTIVE_LOW>;
+
+	spidev@0 {
+		compatible = "max,max31865"; 
+		reg = <0>;
+        spi-max-frequency = <10000000>;		
+	};
+	spidev@1 {
+		compatible = "max,max31865";
+		reg = <1>;
+        spi-max-frequency = <10000000>;
+	};
+};
\ No newline at end of file
-- 
2.25.1

