From a9a723909b9f29e9158cfba684c41583c40f6ecc Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Sat, 29 Oct 2022 20:46:46 +0200
Subject: [PATCH 1/4] Add onewire buses: PC7, PA8, PA9.

---
 .../dts/sun8i-h2-plus-bananapi-m2-zero.dts    | 49 +++++++++++++++++++
 1 file changed, 49 insertions(+)

diff --git a/arch/arm/boot/dts/allwinner/sun8i-h2-plus-bananapi-m2-zero.dts b/arch/arm/boot/dts/allwinner/sun8i-h2-plus-bananapi-m2-zero.dts
index 3963fd58e2f74..569a7933038ce 100644
--- a/arch/arm/boot/dts/allwinner/sun8i-h2-plus-bananapi-m2-zero.dts
+++ b/arch/arm/boot/dts/allwinner/sun8i-h2-plus-bananapi-m2-zero.dts
@@ -109,6 +109,39 @@ wifi_pwrseq: wifi_pwrseq {
 		clocks = <&rtc CLK_OSC32K_FANOUT>;
 		clock-names = "ext_clock";
 	};
+
+	onewire_1 {
+		status = "okay";
+
+		compatible = "w1-gpio";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&onewire_pin_1>;
+		gpios = <&pio 2 7 GPIO_OPEN_DRAIN>;
+		linux,open-drain;
+	};
+
+	onewire_2 {
+		status = "okay";
+
+		compatible = "w1-gpio";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&onewire_pin_2>;
+		gpios = <&pio 0 7 GPIO_OPEN_DRAIN>;
+		linux,open-drain;
+	};
+
+	onewire_3 {
+		status = "okay";
+
+		compatible = "w1-gpio";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&onewire_pin_3>;
+		gpios = <&pio 0 8 GPIO_OPEN_DRAIN>;
+		linux,open-drain;
+	};
 };
 
 &cpu0 {
@@ -297,3 +330,19 @@ spidev@1 {
 	};
 };
 
+&pio {
+	onewire_pin_1: onewire-pin-1 {
+		pins = "PC7";
+		function = "gpio_in";
+	};
+
+	onewire_pin_2: onewire-pin-2 {
+		pins = "PA7";
+		function = "gpio_in";
+	};
+
+	onewire_pin_3: onewire-pin-3 {
+		pins = "PA8";
+		function = "gpio_in";
+	};
+};
-- 
2.25.1

