From f06d811aa8fa237a61f6523f22f58cc9c956e691 Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Sun, 7 May 2023 07:13:46 +0200
Subject: [PATCH] Add WS2812 spidev support

---
 drivers/spi/spidev.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index 08d10f7966bd2..69934aad2b467 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -714,6 +714,7 @@ static const struct spi_device_id spidev_spi_ids[] = {
 	{ .name = "em3581" },
 	{ .name = "si3210" },
 	{ .name = "max31865" },
+	{ .name = "ws2812" },
 	{},
 };
 MODULE_DEVICE_TABLE(spi, spidev_spi_ids);
@@ -743,6 +744,7 @@ static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "silabs,em3581", .data = &spidev_of_check },
 	{ .compatible = "silabs,si3210", .data = &spidev_of_check },
 	{ .compatible = "max,max31865", .data = &spidev_of_check },
+	{ .compatible = "ws,ws2812", .data = &spidev_of_check },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
-- 
2.25.1

